src/mtev_http.c

```diff
@@ -1168,15 +1210,88 @@ mtev_http_session_req_consume_chunked(mtev_http_session_ctx *ctx,
 
   return next_chunk;
 }
+
+/* will uncompress the chain at 'in' and write uncompressed chain at 'out' with
+ * the last member of the out chain in 'last_out'
+ * 
+ * return -1 on error
+ * otherwise return total_uncompressed_size 
+ * */
+static ssize_t
+mtev_http_session_decompress(mtev_compress_type type, struct bchain *in, 
+                             mtev_http_session_ctx *ctx, 
+                             struct bchain **out, struct bchain **last_out)
+{
+  ssize_t total_decompressed_size = 0;
+  mtev_stream_decompress_ctx_t *dctx = ctx->req.decompress_ctx;
+  if (dctx == NULL) {
+    dctx = ctx->req.decompress_ctx = mtev_create_stream_decompress_ctx();
+    if (mtev_stream_decompress_init(dctx, type) < 0) {
+      mtev_destroy_stream_decompress_ctx(dctx);
+      ctx->req.decompress_ctx = NULL;
+      return -1;
+    }
+  }
+
+  if (*out == NULL) {
+    *out = ALLOC_BCHAIN(DEFAULT_BCHAINSIZE);
+  }
+  struct bchain *o = *out;
+  *last_out = *out;
+  while (in && in->size) {
```

Comment:
OK, so I guess the way this is written, this function is only called when the only data in the bchain is compressed, so there's no bug.